### PR TITLE
Set up dokka and docs publishing process

### PR DESCRIPTION
**Pull request by sandwwraith**
Docs can be previewed in my fork: https://sandwwraith.github.io/kotlinx.serialization/kotlinx-serialization/index.html

### CODE REVIEW

File: dokka/moduledoc.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Module kotlinx-serialization-runtime
+
+A runtime library that makes usage of code generated by kotlinx.serialization compiler plugin.
```
Comment (qwwdfsad) [inline dokka/moduledoc.md]:
```suggestion
A companion library to kotlinx.serialization compiler plugin.
```

File: dokka/moduledoc.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+Contains core serialization API, as well as serializers for standard library classes, and ready for use JSON
+format implementation.
```
Comment (qwwdfsad) [inline dokka/moduledoc.md]:
```suggestion
Contains core serialization API, serializers for standard library classes, and ready to use JSON
format implementation.
```

File: formats/properties/commonMain/src/kotlinx/serialization/Properties.kt [hunk 3]
```diff
@@ -10,7 +10,7 @@ import kotlinx.serialization.modules.*
 
 /**
  * Transforms a [Serializable] class' properties into a single flat [Map] which consists of
- * string keys and primitive type values, and vice versa.
+ * string keys and primitive type values, and vice versa. Located in separated `kotlinx-serialization-properties` artifact.
```
Comment (qwwdfsad) [inline formats/properties/commonMain/src/kotlinx/serialization/Properties.kt]: Did you add this mention for the website users?

File: runtime/commonMain/src/kotlinx/serialization/Exceptions.kt [hunk 4]
```diff
@@ -10,12 +10,20 @@ package kotlinx.serialization
 public open class SerializationException(message: String, cause: Throwable? = null) : RuntimeException(message, cause)
 
 // thrown from generated code
+/**
+ * Thrown when [KSerializer] hasn't received property from [Decoder], and this property was not optional.
```
Comment (qwwdfsad) [inline runtime/commonMain/src/kotlinx/serialization/Exceptions.kt]:
```suggestion
 * Thrown when [KSerializer] did not receive property from [Decoder], and this property was not optional.
```

File: runtime/commonMain/src/kotlinx/serialization/SealedSerializer.kt [hunk 5]
```diff
@@ -15,6 +15,8 @@ import kotlin.reflect.*
  * In contrary to [PolymorphicSerializer], all known subclasses with serializers must be passed
  * in `subclasses` and `subSerializers` constructor parameters.
  * If a subclass is a sealed class itself, all its subclasses are registered as well.
+ *
+ * If sealed hierarchy is marked [@Serializable][Serializable], an instance of this class is constructed automatically.
```
Comment (qwwdfsad) [inline runtime/commonMain/src/kotlinx/serialization/SealedSerializer.kt]:
```suggestion
 * If a sealed hierarchy is marked with [@Serializable][Serializable], an instance of this class is provided automatically.
```

File: runtime/commonMain/src/kotlinx/serialization/SerialDescriptorBuilder.kt [hunk 6]
```diff
@@ -74,7 +74,8 @@ public val SerialDescriptor.nullable: SerialDescriptor
     }
 
 /**
- * Builder for [SerialDescriptor].
+ * Builder for [SerialDescriptor] to be used in custom serializers.
```
Comment (qwwdfsad) [inline runtime/commonMain/src/kotlinx/serialization/SerialDescriptorBuilder.kt]:
```suggestion
 * Builder for [SerialDescriptor] for user-defined serializers.
```

File: formats/properties/commonMain/src/kotlinx/serialization/Properties.kt (hunk 3 above)
Comment (sandwwraith) [reply formats/properties/commonMain/src/kotlinx/serialization/Properties.kt:13]: Yes

**Comment by sandwwraith** [comment]
Since `dev` branch is already different from master, documentation will match only next release. Or should I backport this to master and publish 0.20.0 docs?

**Pull request merged**